lib/spack/spack/ci.py

```diff
@@ -1919,32 +1919,48 @@ def reproduce_ci_job(url, work_dir, autostart, gpg_url, runtime):
     job_tags = None
     if "tags" in job_yaml:
         job_tags = job_yaml["tags"]
-        tty.msg("Job ran with the following tags: {0}".format(job_tags))
+        tty.msg(f"Job ran with the following tags: {job_tags}")
 
     entrypoint_script = [
         ["git", "config", "--global", "--add", "safe.directory", mount_as_dir],
-        [".", os.path.join(mount_as_dir if job_image else work_dir, "share/spack/setup-env.sh")],
+        [
+            ".",
+            os.path.join(
+                mount_as_dir if job_image else work_dir,
+                f"share/spack/setup-env.{platform_script_ext}",
+            ),
+        ],
         ["spack", "gpg", "trust", mounted_gpg_path if job_image else gpg_path] if gpg_path else [],
         ["spack", "env", "activate", mounted_env_dir if job_image else repro_dir],
-        [os.path.join(mounted_repro_dir, "install.sh") if job_image else install_script],
+        [
+            (
+                os.path.join(mounted_repro_dir, f"install.{platform_script_ext}")
+                if job_image
+                else install_script
+            )
+        ],
     ]
+    entry_path = os.path.join(mounted_workdir, f"entrypoint.{platform_script_ext}")
+    shell = "powershell.exe" if IS_WINDOWS else "bash"
+    entrypoint = shell if IS_WINDOWS else entry_path
+    exec_cmd = "&" if IS_WINDOWS else "exec"
+    cmd_args = "($args -Join ' ')" if IS_WINDOWS else "$@"
+    no_exit = "-NoExit" if IS_WINDOWS else ""
 
     inst_list = []
     # Finally, print out some instructions to reproduce the build
     if job_image:
         # Allow interactive
+        install_mechanism = (
+            os.path.join(mounted_repro_dir, f"install.{platform_script_ext}")
+            if job_image
+            else install_script
+        )
         entrypoint_script.extend(
             [
-                [
-                    "echo",
-                    "Re-run install script using:\n\t{0}".format(
-                        os.path.join(mounted_repro_dir, "install.sh")
-                        if job_image
-                        else install_script
-                    ),
-                ],
+                ["echo", f"Re-run install script using:\n\t{install_mechanism}"],
                 # Allow interactive
-                ["exec", "$@"],
+                [exec_cmd, cmd_args, no_exit],
```

Comment:
I have a similar request here as for `process_command`: since each of these 3 elements individually changes based on Windows/not-Windows, can you create this like:

```
if IS_WINDOWS:
  value = [
    "&", "($args -Join ' ')", "-NoExit"
  ]
else:
  value = [...]
```